packages/reference/src/components/LinkActions/LinkEntityActions.tsx

```diff
@@ -64,29 +64,38 @@ export function useLinkActionsProps(props: LinkEntityActionsProps): LinkActionsP
     [sdk, entityType, props.onCreate, props.onAction]
   );
 
-  const onLinkExisting = React.useCallback(async () => {
-    const entity = await selectSingleEntity({ sdk, entityType, validations });
-    if (!entity) {
-      return;
-    }
-    props.onLink([entity.sys.id]);
-    props.onAction &&
-      props.onAction({ type: 'select_and_link', entity: entityType, entityData: entity });
-  }, [sdk, entityType, props.onLink, props.onAction]);
+  const onLinkExisting = React.useCallback(
+    async (index?: number) => {
+      const entity = await selectSingleEntity({ sdk, entityType, validations });
+      if (!entity) {
+        return;
+      }
+      props.onLink([entity.sys.id], index);
```

Comment:
This diff is ugly because of the linter but the only thing to take note of is we're now parameterizing the optional index and delegating it to `onLink`.